contracts/Governance/TransactionRelay.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract TransactionRelay is Ownable {
+    event CreateTransaction(
+        bytes32 indexed txHash,
+        address indexed creator,
+        address indexed targetContractAddress,
+        uint256 value,
+        string signature,
+        bytes data
+    );
+    event DeleteTransaction(
+        bytes32 indexed txHash,
+        address indexed creator,
+        address indexed targetContractAddress,
+        uint256 value,
+        string signature,
+        bytes data
+    );
+    event ExecuteTransaction(
+        bytes32 indexed txHash,
+        address indexed executioner,
+        address creator,
+        address indexed targetContractAddress,
+        uint256 value,
+        string signature,
+        bytes data,
+        bytes returnData
+    );
+
+    mapping(bytes32 => StoredTransaction) public storedTransactions;
+
+    struct StoredTransaction {
+        address creator;
+        address targetContractAddress;
+        uint256 value;
+        string signature;
+        bytes data;
+    }
+
+    constructor() public {}
+
+    function createTransaction(
+        address targetContractAddress,
+        uint256 value,
+        string memory signature,
+        bytes memory data
+    ) external returns (bytes32 txHash) {
+        txHash = keccak256(abi.encode(msg.sender, targetContractAddress, value, signature, data));
+
+        if (storedTransactions[txHash].creator == address(0)) {
```

Comment:
`storedTransactions[txHash].creator` will not throw an error if `storedTransactions[txHash]` is empty?

Reply:
@tkeidar nope, `storedTransactions[txHash]` will be all zeros and therefore `storedTransactions[txHash].creator` will be `0` = `address(0)`